### PR TITLE
fix(core/react): fix connection issues with connection-checker, avoid resending duplicates

### PR DESCRIPTION
**Pull request by vanbasten17**
## Description
* Hotfix PR (published alpha) trying to fix some issues regarding connection.
* I have opted for using `connection-checker` library instead of relying on `navigator.onLine` (could not work for all browsers) adapting the current `useNetwork` hook.
* Bumped alpha version
* Bumped versions to 0.18.1

### CODE REVIEW

File: packages/botonic-react/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@botonic/react",
-  "version": "0.18.0",
+  "version": "0.18.1",
```
Comment (asastre) [inline packages/botonic-react/package.json:3]: Since this has been published as alpha, should the new versions contain 'alpha' too?

**Review by asastre — APPROVED** [review]
LGTM!

File: packages/botonic-react/package.json (hunk 1 above)
Comment (vanbasten17) [reply packages/botonic-react/package.json:3]: if the PR is approved I will publish the 0.18.1 version

**Review by dpinol — APPROVED** [review]

**Comment by vanbasten17** [comment]
@asastre , @dpinol just added a pair of tests before publishing. See commit: https://github.com/hubtype/botonic/pull/1302/commits/ea80ebaccc2385340a988170a0dbeb7ac5070aa5

**Pull request merged**